src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -912,7 +912,12 @@ private void WriteLocalFileHeaderAndDataIfNeeded()
                 {
                     // we know the sizes at this point, so just go ahead and write the headers
                     if (_uncompressedSize == 0)
+                    {
+                        // according to ZIP specs, zero-byte files MUST NOT include file data
+                        _compressedSize = 0;
                         CompressionMethod = CompressionMethodValues.Stored;
+                    }
+                        
                     WriteLocalFileHeader(isEmptyFile: false);
```

Comment:
How does this `isEmptyFile` argument relate?  Should this be true? And if yes, does `_compressedSize` need to be change to 0?

Reply:
My bad thanks for spotting it out! 
`_compressedSize` needs to be changed because it will be used later on at the central directory, I added a comment.